jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -123,29 +134,97 @@ public List<MappedResource<E>> getMatches(String path)
     {
         boolean isRootPath = "/".equals(path);
 
-        List<MappedResource<E>> ret = new ArrayList<>();
+        if (_mappings.isEmpty())
```

Comment:
Could actually move this test to the first line of the method.

Reply:
done